### DISCUSSION TITLE
Context insertion for auto-instrumented spans

### DISCUSSION THREAD

**Discussion by jagraj-singh** [aspecto-io/opentelemetry-ext-js#178] [Q&A] [answered]
//span created here using AWS autoinstrumentation
const data = await this.SqsService.sendMessage(
      JSON.stringify(message),
      'https://sqs.us-east-1.amazonaws.com/account_number/telemetry',
);

I am thinking to insert the context of auto instrumented span in the SQS message using preRequestHook. Am I thinking correctly..?

**Comment by blumamir** [comment, marked as answer]
Hi @Jagraj02 
Can you elaborate on what you are trying to achieve?
The context is already injected into the message attributes and propagated to the receiver.

**Reply by jagraj-singh** [reply]
The issue I am facing is that the receiver span seems to be completely unrelated to the producer.

![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/40578779/131840792-59371dfe-6a7d-4b5a-8a8d-ec92c5b38fc5.png)

Is this an expected behavior..? Do I have to manage context by reading message attributes individually for every message..?

**Reply by blumamir** [reply]
The span you posted above is the `messaging.operation: receive` span. The context should be set on the `messaging.operation: process` span as link. This behavior follows the [semantic conventions for batch receive on messaging systems](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#batch-receiving).

The implementation in JS is not complete due to the way aws-sdk api is implemented. you can read [here](https://github.com/aspecto-io/opentelemetry-ext-js/blob/master/packages/instrumentation-aws-sdk/docs/sqs.md#processing-spans) on what are the limitations and how to debug the situation (resource can be probably improved, please share feedback if it's clear enough or not).

If you are unable to resolve the issue, please comment here with a snippet of the code that process the sqs messages in your application (how you loop on the Messages array)

**Reply by jagraj-singh** [reply]
Thanks, @blumamir I get your explanation and also I am clear now why the receive span isn't using the context.
I am also able to use the message attributes in the set context in the process span.
